velox/exec/tests/utils/TpchQueryBuilder.h

```diff
@@ -0,0 +1,125 @@
+/// This information is inferred from the input data files.
+struct TpchTableMetadata {
+  RowTypePtr type;
+  std::vector<std::string> dataFiles;
+  std::unordered_map<std::string, std::string> columnAliases;
+};
+
+/// Builds TPC-H queries using TPC-H data files located in the specified
+/// directory. Each table data must be placed in hive-style partitioning. That
+/// is, the top-level directory is expected to contain a sub-directory per table
+/// name and the name of the sub-directory must match the table name. Example:
+/// ls -R data/
+///  customer   lineitem
+///
+///  data/customer:
+///  customer1.parquet  customer2.parquet
+///
+///  data/lineitem:
+///  lineitem1.parquet  lineitem2.parquet  lineitem3.parquet
+
+/// The column names can vary. Additional columns may exist towards the end.
+/// The class uses standard names (example: l_returnflag) to build TPC-H plans.
+/// Since the column names in the file can vary, they are mapped to the standard
+/// names. Therefore, the order of the columns in the file is important and
+/// should be in the same order as in the TPC-H standard.
+class TpchQueryBuilder {
+ public:
+  TpchQueryBuilder(dwio::common::FileFormat format) : format_(format) {}
+
+  /// Read each data file, initialize row types, and determine data paths for
+  /// each table.
+  /// @param dataPath path to the data files
+  void initialize(const std::string& dataPath);
+
+  /// Get the query plan for a given TPC-H query number.
+  /// @param queryId TPC-H query number
+  TpchPlan getQueryPlan(int queryId) const;
+
+  /// Get the TPC-H table names present.
+  static const std::vector<std::string>& getTableNames();
+
+ private:
+  TpchPlan getQ1Plan() const;
+  TpchPlan getQ6Plan() const;
+
+  const std::vector<std::string>& getTableFilePaths(
+      const std::string& tableName) const {
+    return tableMetadata_.at(tableName).dataFiles;
+  }
+
+  const std::string& getColumnAlias(
+      const std::string& tableName,
+      const std::string& columnName) const {
+    return tableMetadata_.at(tableName).columnAliases.at(columnName);
+  }
+
+  std::vector<std::string> getColumnAliases(
+      const std::string& tableName,
+      const std::vector<std::string>& columnNames) const {
+    std::vector<std::string> aliases;
+    for (const auto& name : columnNames) {
+      aliases.push_back(getColumnAlias(tableName, name));
+    }
+    return aliases;
+  }
+
+  std::vector<std::string> getProjectColumnAliases(
+      const std::string& tableName,
+      const std::vector<std::string>& columnNames) const {
+    std::vector<std::string> aliases;
+    for (const auto& name : columnNames) {
+      aliases.push_back(getColumnAlias(tableName, name) + " AS " + name);
+    }
+    return aliases;
+  }
+
+  std::unordered_map<std::string, TpchTableMetadata> tableMetadata_;
+  const dwio::common::FileFormat format_;
+  static const std::unordered_map<std::string, std::vector<std::string>>
```

Comment:
Got another linter error for these statics:

Non POD static variables are susceptible to static init order bugs (see https://isocpp.org/wiki/faq/ctors#static-init-order). Even if the constructor called by the static variable currently does not depend on other static objects, that may not always hold in the future. The initialization order of static variables is non-deterministic for variables defined in different .cpp files, which means this may one day manifest into a difficult to diagnose heisenbug if/when we add a dependency on another static variable. See t8565728.
For objects without a `constexpr` constructor you can use:
```
static const Foo& getFoo() {
  static const Foo foo = <something>;
  return foo;
}
```

Reply:
Good to know. Thank you for handling this.